keras_nlp/evaluations/glue/glue.py

```diff
@@ -0,0 +1,287 @@
+
+    labelname = labelnames.get(FLAGS.task_name)
+    test_iterator = iter(test_ds)
+    with tf.io.gfile.GFile(filename, "w") as f:
+        # GLUE requires a format of index + tab + prediction.
+        writer = csv.writer(f, delimiter="\t")
+        # Write the required headline for GLUE.
+        writer.writerow(["index", "prediction"])
+        for i in range(test_ds.cardinality()):
+            # TODO(chenmoneygithub): Add distribution strategy support.
+            pred = eval_step(test_iterator)
+            pred = pred.numpy()
+            for j in range(len(pred)):
+                idx = i * FLAGS.batch_size + j
+                if labelname:
+                    pred_value = labelname[int(pred[j])]
+                else:
+                    pred_value = pred[j]
+                writer.writerow([idx, pred_value])
+            break
+
+
+@keras.utils.register_keras_serializable(package="custom")
+class GlueClassifier(keras.Model):
+    """Default classification model for GLUE tasks.
+
+    `GlueClassifier` takes in a pretrained model as `backbone`, which must
+    output a single vector representation per input data. If the pretrained
+    model outputs a dictionary, users need to make a wrapper model to
+    extract out the representation.
+
+
+    Args:
+        backbone: A keras model instance. The output must be a single vector
+            representation per input data.
+        num_classes: int, the number of classes to predict.
+    """
+
+    def __init__(
+        self,
+        backbone,
+        num_classes=2,
+    ):
+        inputs = backbone.input
+        representation = backbone(inputs)
+        outputs = keras.layers.Dense(
+            num_classes,
+            kernel_initializer=keras.initializers.TruncatedNormal(stddev=0.02),
+            name="logits",
+        )(representation)
+        # Instantiate using Functional API Model constructor.
+        super().__init__(inputs=inputs, outputs=outputs)
+        # All references to `self` below this line.
+        self.backbone = backbone
+        self.num_classes = num_classes
+
+    def get_config(self):
+        return {
+            "backbone": keras.layers.serialize(self.backbone),
+            "num_classes": self.num_classes,
+        }
+
+    @classmethod
+    def from_config(cls, config):
+        if "backbone" in config:
+            config["backbone"] = keras.layers.deserialize(config["backbone"])
+        return cls(**config)
+
+
+def main(_):
+    train_ds, test_ds, val_ds = load_data(FLAGS.task_name)
+
+    # ----- Custom code block starts -----
+    bert_model = keras_nlp.models.Bert.from_preset("bert_tiny_uncased_en")
+    bert_preprocessor = keras_nlp.models.BertPreprocessor.from_preset(
+        "bert_tiny_uncased_en"
+    )
+
+    def preprocess_fn(feature, label):
+        return bert_preprocessor(feature), label
+
+    # ----- Custom code block ends -----
+
+    train_ds = preprocess_data(preprocess_fn, train_ds)
```

Comment:
After https://github.com/keras-team/keras-nlp/pull/478, this can just be `train_ds.map(preprocessor).batch(FLAGS.batch_size).prefetch(tf.data.AUTOTUNE)`

(though no need to wait for that PR.